### PR TITLE
Feat/136 simple graph segmentation for damages

### PR DESCRIPTION
**Pull request by sahand-asgarpour**
## Issue addressed
Solves #136 

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
Explain how you addressed the resolution of the related issue, what choices you made and why.

### Checklist
- [x] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [ ] Updated documentation if needed.

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: ra2ce/network/network_wrappers/vector_network_wrapper.py [hunk 1]
```diff
@@ -111,6 +114,17 @@ def get_network(
         edges_complex, _ = nut.graph_to_gdf(graph_complex)
         logging.info("Finished converting the graph to a geodataframe")
 
+        # Segment the complex graph
+        if not math.isnan(self.segmentation_length):
```
Comment (MatthiasHauthDeltares) [inline ra2ce/network/network_wrappers/vector_network_wrapper.py:118]: This is repetition from what happens in OSMNetworkWrapper. Make it a function and use it there

Comment (sahand-asgarpour) [reply ra2ce/network/network_wrappers/vector_network_wrapper.py]: Ok, I make a function at the protocol level.

**Review by MatthiasHauthDeltares — APPROVED** [review]

**Comment by ArdtK** [comment]
@sahand-asgarpour in `NetworkWrapperProtocol` you added method `segment_graph`. It is not correct to add implemented methods to a protocol. A protocol is meant to only contain signatures (definitions).
If you would like to have such a method that is used in all implementing classes, you should create a base class next to the protocol.

**Comment by sahand-asgarpour** [comment]
> @sahand-asgarpour in `NetworkWrapperProtocol` you added method `segment_graph`. It is not correct to add implemented methods to a protocol. A protocol is meant to only contain signatures (definitions). If you would like to have such a method that is used in all implementing classes, you should create a base class next to the protocol.

I understand. Thank you for the clarification. I make an issue for that to be picked up later. Is it an idea?

**Comment by Carsopre** [comment]
This PR should have not been merged when there's a configuration build failing!

**Pull request merged**